src/vs/workbench/parts/preferences/browser/preferencesRenderers.ts

```diff
@@ -531,6 +564,244 @@ export class HiddenAreasRenderer extends Disposable {
+		const settingsTextEndLine = settingsTextStartLine + totalLines - 1;
+		this.editor.executeEdits(MostRelevantMatchesRenderer.editId, [{
+			text: settingsValue,
+			forceMoveMarkers: false,
+			range: new Range(settingsTextStartLine, 0, settingsTextEndLine, 0),
+			identifier: { major: 1, minor: 0 }
+		}]);
+
+		this.editor.updateOptions({ readOnly: true });
+
+		return settingsTextEndLine;
+	}
+
+	private renderSearchResultsSection(): void {
+		this.editor.updateOptions({ readOnly: false });
+
+		this.editor.executeEdits(MostRelevantMatchesRenderer.editId, [{
+			text: MostRelevantMatchesRenderer.bunchOfNewlines,
+			forceMoveMarkers: false,
+			range: new Range(MostRelevantMatchesRenderer.settingsInsertStart, 0, MostRelevantMatchesRenderer.settingsInsertEnd + 1, 0),
+			identifier: { major: 1, minor: 0 }
+		}]);
+
+		this.editor.updateOptions({ readOnly: true });
+	}
+
+	private getOrderedSettingRanges(filteredGroups: ISettingsGroup[], allSettingsGroups: ISettingsGroup[], scores: any, model: editorCommon.IModel): IRange[] {
+		// Manually exclude mostCommonlyUsed section to avoid dupes
+		if (filteredGroups.length && filteredGroups[0].id === 'mostCommonlyUsed') {
+			filteredGroups.shift();
+		}
+
+		const matchingRanges: { range: IRange, name: string }[] = [];
+		for (const group of allSettingsGroups) {
+			const filteredGroup = filteredGroups.filter(g => g.title === group.title)[0];
+			if (filteredGroup) {
+				for (const section of group.sections) {
+					for (const setting of section.settings) {
+						if (this.containsLine(setting.range.startLineNumber, filteredGroup)) {
+							matchingRanges.push({
+								name: setting.key,
+								range: this.createCompleteRange(setting.range, model)
+							});
+						}
+					}
+				}
+			}
+		}
+
+		return matchingRanges
+			.sort((a, b) => scores[b.name] - scores[a.name])
+			.map(r => r.range);
+	}
+
+	private containsLine(lineNumber: number, settingsGroup: ISettingsGroup): boolean {
+		if (settingsGroup.titleRange && lineNumber >= settingsGroup.titleRange.startLineNumber && lineNumber <= settingsGroup.titleRange.endLineNumber) {
+			return true;
+		}
+
+		for (const section of settingsGroup.sections) {
+			if (section.titleRange && lineNumber >= section.titleRange.startLineNumber && lineNumber <= section.titleRange.endLineNumber) {
+				return true;
+			}
+
+			for (const setting of section.settings) {
+				if (lineNumber >= setting.range.startLineNumber && lineNumber <= setting.range.endLineNumber) {
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+	private createCompleteRange(range: IRange, model: editorCommon.IModel): IRange {
+		return {
+			startLineNumber: range.startLineNumber,
+			startColumn: model.getLineMinColumn(range.startLineNumber),
+			endLineNumber: range.endLineNumber,
+			endColumn: model.getLineMaxColumn(range.endLineNumber)
+		};
+	}
+}
+
+export class FeedbackWidgetRenderer extends Disposable {
```

Comment:
Do you think if it is a good idea to provide some `arrow` actions next to results to update the ordering? Also listen to user actions like copying the setting after searching and update the setting relevance to the search criteria?

I see the necessary of feedback, but IMO if we can get more details from normal user actions would be more helpful. Not sure how many users update the contents ordering using the feedback widget as it looks to me a cumbersome task.

Reply:
I think for real users, an 'arrow' action would be more practical, although I couldn't decide how it should look. This mechanism with opening an editor is really for the team's dev purposes.